library/core/src/alloc/layout.rs

```diff
@@ -372,38 +372,46 @@ impl Layout {
     /// padding is inserted, the alignment of `next` is irrelevant,
     /// and is not incorporated *at all* into the resulting layout.
     ///
-    /// On arithmetic overflow, returns `LayoutErr`.
+    /// On arithmetic overflow, returns `LayoutError`.
     #[unstable(feature = "alloc_layout_extra", issue = "55724")]
     #[inline]
-    pub fn extend_packed(&self, next: Self) -> Result<Self, LayoutErr> {
-        let new_size = self.size().checked_add(next.size()).ok_or(LayoutErr { private: () })?;
+    pub fn extend_packed(&self, next: Self) -> Result<Self, LayoutError> {
+        let new_size = self.size().checked_add(next.size()).ok_or(LayoutError { private: () })?;
         Layout::from_size_align(new_size, self.align())
     }
 
     /// Creates a layout describing the record for a `[T; n]`.
     ///
-    /// On arithmetic overflow, returns `LayoutErr`.
+    /// On arithmetic overflow, returns `LayoutError`.
     #[stable(feature = "alloc_layout_manipulation", since = "1.44.0")]
     #[inline]
-    pub fn array<T>(n: usize) -> Result<Self, LayoutErr> {
+    pub fn array<T>(n: usize) -> Result<Self, LayoutError> {
         let (layout, offset) = Layout::new::<T>().repeat(n)?;
         debug_assert_eq!(offset, mem::size_of::<T>());
         Ok(layout.pad_to_align())
     }
 }
 
+#[stable(feature = "alloc_layout", since = "1.28.0")]
+#[rustc_deprecated(
+    since = "1.51.0",
+    reason = "use LayoutError instead",
+    suggestion = "LayoutError"
+)]
+pub type LayoutErr = LayoutError;
```

Comment:
Unfortunately I don't think the deprecated attributes actually work on type aliases (or re-exports as far as I know).

Maybe we could build a local distribution or add a ui test and confirm.

Reply:
I did some local testing, and they do work on type aliases, but not on reexports. 

I found the following UI tests covering type alias deprecation:
https://github.com/rust-lang/rust/blob/master/src/test/ui/stability-attribute/generics-default-stability.rs#L161-L180 (turns out I wrote these ones :) )
https://github.com/rust-lang/rust/blob/master/src/test/ui/deprecation/deprecation-lint-nested.rs#L55

However it seems that none of the stability attributes have an effect on reexports:
https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=bd79ebede5635e21719b5439a26e0cc5

Despite this, there is precedence for placing `#[rustc_deprecated]` on reexports:
https://github.com/rust-lang/rust/blob/master/library/core/src/intrinsics.rs#L66-L70
https://github.com/rust-lang/rust/blob/master/library/std/src/collections/mod.rs#L404-L406

Reply:
Thanks for checking into this @exrook! So it's just the re-export that won't raise deprecated warnings. I'm ok keeping the `deprecated` attribute on the re-export too as a signal even though it doesn't actually do anything. Anybody `use`ing that re-export will still get a deprecation warning.